ignite/pkg/cosmosanalysis/app/app.go

```diff
@@ -144,37 +154,25 @@ func findBasicManagerRegistrations(n ast.Node, basicManagerModule string) []stri
 		return nil
 	}
 
-	packagesRegistered := make([]string, len(callExprType.Args))
-	for i, arg := range callExprType.Args {
-		argAsCompositeLitType, ok := arg.(*ast.CompositeLit)
-		if ok {
-			compositeTypeSelectorExpr, ok := argAsCompositeLitType.Type.(*ast.SelectorExpr)
-			if !ok {
-				continue
-			}
-
-			compositeTypeX, ok := compositeTypeSelectorExpr.X.(*ast.Ident)
-			if ok {
-				packagesRegistered[i] = compositeTypeX.Name
-				continue
-			}
-		}
-
-		argAsCallType, ok := arg.(*ast.CallExpr)
-		if ok {
-			argAsFunctionType, ok := argAsCallType.Fun.(*ast.SelectorExpr)
-			if !ok {
-				continue
-			}
-
-			argX, ok := argAsFunctionType.X.(*ast.Ident)
-			if ok {
-				packagesRegistered[i] = argX.Name
-			}
+	var packages []string
+	for _, arg := range callExprType.Args {
+		switch v := arg.(type) {
+		case *ast.CompositeLit:
+			// The arg is an app module
+			packages = append(packages, parsePkgNameFromCompositeLit(v))
+		case *ast.CallExpr:
+			// The arg is a function call that returns the app module
+			packages = append(packages, parsePkgNameFromCall(v))
+		case *ast.Ident:
+			// The list of modules are defined in a local variable
+			packages = append(packages, parseAppModulesFromIdent(v, pkgDir)...)
+		case *ast.SelectorExpr:
+			// The list of modules is defined in a variable of a different package
+			packages = append(packages, parseAppModulesFromSelectorExpr(v, pkgDir, pkgs)...)
```

Comment:
```suggestion
			packages = append(packages, parseAppModulesFromSelectorExpr(v, pkgDir, pkgs)...)
		default:
			panic(fmt.Sprintf("unhandled type %v", v))			
```

What do you think of panic-ing if the type is not handled ?
It's less user friendly but on the other hand, it will help us to improve the code analysis when the user will report it, and it's probably better than reporting that everything went well whereas some modules are missing in the ts-client.

Reply:
I thought about a refactor to handle errors properly while implementing the code in this section because as you provably saw, the way the AST inspection is implemented doesn't support errors. I will change it to use panics and then recover to generate a proper error message instead of a traceback. It is not the most elegant solution but it seems to be the right one for the current implementation.

Thanks for the recommendation @tbruyelle!

Reply:
Finally I implemented error support without the need of deferred and recover calls in 072521d